Source/DafnyLanguageServer.Test/Lookup/DefinitionTest.cs

```diff
@@ -28,20 +25,265 @@ public class DefinitionTest : ClientBasedLanguageServerTest {
+      Assert.AreEqual(documentItem.Uri, usizeReference.Location.Uri);
+      Assert.AreEqual(new Range((2, 7), (2, 12)), usizeReference.Location.Range);
+
+      var lengthDefinition = (await RequestDefinition(documentItem, (9, 10)).AsTask());
+      Assert.IsFalse(lengthDefinition.Any());
+
+      var providerImport = (await RequestDefinition(documentItem, (6, 16)).AsTask()).Single();
+      Assert.AreEqual(new Range((0, 17), (0, 25)), providerImport.Location!.Range);
+
+      var lengthAssignment = (await RequestDefinition(documentItem, (10, 7)).AsTask()).Single();
+      Assert.AreEqual(new Range((9, 8), (9, 14)), lengthAssignment.Location!.Range);
+    }
+
+    [TestMethod]
+    public async Task JumpToOtherModule() {
+      var source = @"
+module Provider {
+  class A {
+    var x: int;
+
+    constructor() {}
+
+    function method GetX(): int
+      reads this`x
+    {
+      this.x
+    }
+  }
+}
+
+module Consumer {
+  import opened Provider
+
+  method DoIt() returns (x: int) {
+    var a := new A();
+    return a.GetX();
+  }
+}
+
+module Consumer2 {
+  import Provider
+
+  type A2 = Provider.A
+}".TrimStart();
+      var documentItem = CreateTestDocument(source);
+      await client.OpenDocumentAndWaitAsync(documentItem, CancellationToken);
+      var getXCall = (await RequestDefinition(documentItem, (19, 13)).AsTask()).Single();
+      Assert.AreEqual(new Range((6, 20), (6, 24)), getXCall.Location!.Range);
+
+      var xInFrame = (await RequestDefinition(documentItem, (7, 17)).AsTask()).Single();
+      Assert.AreEqual(new Range((2, 8), (2, 9)), xInFrame.Location!.Range);
+
+      var providerQualifier = (await RequestDefinition(documentItem, (26, 14)).AsTask()).Single();
+      Assert.AreEqual(new Range((24, 9), (24, 17)), providerQualifier.Location!.Range);
+    }
+
+    [TestMethod]
+    public async Task DefinitionOfMethodInvocationOfMethodDeclaredInSameDocumentReturnsLocation() {
+      var source = @"
+module Container {
+  method GetIt() returns (x: int) {
+  }
+
+  method DoIt(arg: int) {
+  }
+}
+
+method CallIts() returns () {
+  var x := Container.GetIt();
+  Container.DoIt(x);
+}".TrimStart();
+      var documentItem = CreateTestDocument(source);
+      await client.OpenDocumentAndWaitAsync(documentItem, CancellationToken);
+
+      var containerReference = (await RequestDefinition(documentItem, (9, 11)).AsTask()).Single();
+      Assert.AreEqual(new Range((0, 7), (0, 16)), containerReference.Location!.Range);
+
+      var getItCall = (await RequestDefinition(documentItem, (9, 23)).AsTask()).Single();
+      Assert.AreEqual(new Range((1, 9), (1, 14)), getItCall.Location!.Range);
+
+      var doItCall = (await RequestDefinition(documentItem, (10, 12)).AsTask()).Single();
+      Assert.AreEqual(new Range((4, 9), (4, 13)), doItCall.Location!.Range);
+
+      var xVar = (await RequestDefinition(documentItem, (10, 17)).AsTask()).Single();
+      Assert.AreEqual(new Range((9, 6), (9, 7)), xVar.Location!.Range);
     }
```

Comment:
Is there a way to jump to another file? If so, can we test this?

Reply:
It is tested in `DefinitionOfFunctionInvocationOfFunctionDeclaredInForeignDocumentReturnsLocation`

Reply:
I'm sorry, I cannot find this function here. Did it exist in the past? If so, last time I checked, it's not working on my machine.

Reply:
It's here: https://github.com/dafny-lang/dafny/blob/a2b69626b9d17a648e6c494ce17a5f8fa31bbc51/Source/DafnyLanguageServer.Test/Lookup/DefinitionTest.cs#L313